x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/read_rules_route.ts

```diff
@@ -41,8 +44,24 @@ export const createReadRulesRoute: Hapi.ServerRoute = {
         id,
         ruleId,
       });
+      const ruleStatuses = await savedObjectsClient.find({
+        type: ruleStatusSavedObjectType,
+        perPage: 10,
+        search: `"${id}"`,
+        searchFields: ['alertId'],
+      });
+      ruleStatuses.saved_objects.sort((a, b) => {
```

Comment:
For the searches once you move over to date times you can do:

```ts
        const ruleStatuses = await savedObjectsClient.find({
          type: ruleStatusSavedObjectType,
          perPage: 5,
          search: `"${rule.id}"`,
          searchFields: ['alertId'],
          sortField: 'statusDate',
          sortOrder: 'desc',
        });
```
Where you're using `sortOder: 'desc'` and `sortField` and the `perPage` if you set it to 5 then it will return you 5 records and you will have everything you need.